### PR TITLE
Add instance tests

### LINKED ISSUE #2080: 2.x->3.0 Test (29-instance.sh)
Port test 29-instance.sh from 2.6 shell script to 3.0's new golang framework.

### PR DESCRIPTION
**Pull request by jscook2345**
**Description of the Pull Request (PR):**

Add basic instance test that builds an nginx image bootstrapped from docker, runs instance start, instance list, and instance stop.

**This fixes or addresses the following GitHub issues:**

- Fixes #2080

**Checkoff for all PRs:**

- [X] I have read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [user](https://www.github.com/sylabs/singularity-userdocs) and [administrator](https://www.github.com/sylabs/singularity-admindocs) documentation bases.
- [X] I have tested this PR locally with a `make testall`
- [X] This PR is against the project's master branch
- [ ] I have added myself as a contributor to the [contributors's file](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)
- [X] This PR is ready for review and/or merge

Attn: @singularity-maintainers

### CODE REVIEW

File: src/cmd/singularity/instance_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		}
+
+		listInstanceOutput, listInstanceError := listInstance()
+
+		if listInstanceError != nil {
+			t.Fatalf("Error listing instances: %v. Output follows.\n%s", listInstanceError, string(listInstanceOutput))
+		}
+
+		stopInstanceOutput, stopInstanceError := stopInstance(defaultInstance)
+
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/instance_test.go]: I would rm this white line

File: src/cmd/singularity/instance_test.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	if buildImageError != nil {
+		t.Fatalf("Error building an image from a definition: %v. Output follows.\n%s", buildImageError, string(buildImageOutput))
+	}
+
+	t.Run("StartListStop", test.WithoutPrivilege(func(t *testing.T) {
+		var defaultInstance = "www"
+
+		startInstanceOutput, startInstanceError := startInstance(defaultImage, defaultInstance)
+
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/instance_test.go]: I would rm this white line

File: src/cmd/singularity/instance_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+		if startInstanceError != nil {
+			t.Fatalf("Error starting instance from an image: %v. Output follows.\n%s", startInstanceError, string(startInstanceOutput))
+		}
+
+		listInstanceOutput, listInstanceError := listInstance()
+
```
Comment (ArangoGutierrez) [inline src/cmd/singularity/instance_test.go]: I would rm this white line

**Review by ArangoGutierrez — APPROVED** [review]
LGTM just some minor nipticks

**Review by jmstover — APPROVED** [review]

**Pull request merged**